File: jimmer-example/spring-boot-with-jimmer-multiple-datasource/src/main/java/org/jimmer/jimmerdatasouce/domain/Author.java
package org.jimmer.jimmerdatasouce.domain;

import jakarta.annotation.Nullable;
import org.babyfish.jimmer.sql.*;

import java.time.LocalDateTime;

/**
 * @author jhlz
 * @version x.x.x
 */
@Entity
@Table(name = "db_author")
public interface Author {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id();

    String username();

    int age();

    String gender();

    @Nullable
    String address();

    LocalDateTime createTime();
    LocalDateTime updateTime();
    String createBy();
    @Nullable
    String updateBy();
}

File: jimmer-example/quarkus-with-jimmer-kt/src/main/kotlin/org/jimmer/domain/Author.kt
package org.jimmer.domain

import org.babyfish.jimmer.sql.*

/**
 * @author jhlz
 * @version x.x.x
 */
@Entity
@Table(name = "db_author")
interface Author: BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    @Key
    val username: String

    val age: Int

    val gender: String

    val address: String?

    @ManyToMany
    @JoinTable(
        name = "author_book",
        joinColumnName = "author_id",
        inverseJoinColumnName = "book_id"
    )
    val books: List<Book>
}